packages/aws-cdk/test/util/stack-monitor.test.ts

```diff
@@ -124,11 +124,20 @@ describe('stack monitor, collecting errors from events', () => {
         return {
           StackEvents: [
             addErrorToStackEvent(
-              event(100), {
+              event(102), {
                 logicalResourceId: 'nestedStackLogicalResourceId',
                 physicalResourceId: 'nestedStackPhysicalResourceId',
                 resourceType: 'AWS::CloudFormation::Stack',
                 resourceStatusReason: 'nested stack failed',
+                resourceStatus: 'UPDATE_FAILED',
+              },
+            ),
+            addErrorToStackEvent(
```

Comment:
Yeah. The code that polls for CloudFormation events and determines whether certain events are errors, etc, was locked up in the activity monitor. It would only periodically poll and then print the results.

I also wanted it on-demand for the `--force` (`--orphan-all`) feature, so I pulled it out into a separate class.